### ISSUE TITLE
Webpack dev server live reload doesn't work with SSH connection

### ISSUE THREAD

**Issue by ralphsmith80** [microsoft/vscode-remote-release#2371]
During web development a common flow it so run a development server in watch mode that will reload or use hot module reload when a file is changed and saved. 

There is a number of ways to set this up. This is just one simplistic way to test the use case.

1. Create a VM with Node
1. Connect vscode to VM using SSH method
1. Create a React app with create-react-app - `npx create-react-app test-app`
1. `cd test-app`
1. `npm i` - install dependencies
1. `npm start` - start development server
1. Open server URL in a browser
1. Open `src/App.js`
1. Make a change
1. Save the file

Expected:
Webpack server should recompile and page should reload

Actual:
Webpack dev server doesn't recompile and changes are not reflect in the browser webpage

The only way to get the changes to show up is to stop and restart the development server. This is very inefficient. 
As a sanity check if I use a terminal to ssh in to the VM without using vscode remote connection. I can start the development server and use vim to make a file change. The Webpack server will recompile and live reload as expected. 

The is so close to enabling me to move to a remote VM workflow that I like.

**Comment by chrmarti** [comment]
@bpasero How is writing the file from VS Code different from writing it from VIM? This sounds like it should not be affected by VS Code itself.

**Comment by bpasero** [comment]
@chrmarti I am not sure how I can answer that. They probably both use the OS native methods for file access, but how would I know without looking into VIM's source code?

Our implementation is [here](https://github.com/microsoft/vscode/blob/ee10caba68179e55ec51e66db2563d3ba07ad180/src/vs/platform/files/node/diskFileSystemProvider.ts#L165) and uses node.js methods for `open`, `write`, `close`.

**Comment by ralphsmith80** [comment]
It looks like it might depend on some levels of complexity. How complex? I have no clue yet. 

This very simple [webpack-demo](https://github.com/ralphsmith80/webpack-demo) worked, but my production level application doesn't work. It's based on [create-react-app](https://github.com/facebook/create-react-app) which uses webpack under the hood so I wouldn't expect that to make much difference. 

I'm out of time for tonight but I'll try this with a brand new create-react-app with nothing added next.

**Comment by 2lazy2debug** [comment]
@ralphsmith80 I'm having the same issue but with standalone webpack and while I don't have a solution, a workaround that worked for me was to run webpack --watch by sshing  in my case and only then fire up vscode.

**Comment by ralphsmith80** [comment]
Well this is working now. 

I create a new create-react-app app and run into an inotify-watchers when trying to start the app. 

There's a related facebook/create-react-app#7612 and the [suggested fix](https://github.com/guard/listen/wiki/Increasing-the-amount-of-inotify-watchers#the-technical-details) worked to get that running.

In that simple create-react-app it reloaded as expected so I tried my production repository again.

On the first file change I received a notification in vscode. I don't recall the message because it didn't sound relevant at the time and I didn't know if it was already present. The server (website) became unresponsive. I just waited. When I refreshed the page the change was picked up. 

After that first hiccup it worked as expected. There seems to be a bit of a lag but I can't say if that will be noticeable when actually coding. 

Best guess at this point is that it was to do the number of inotify watchers. However, I don't know why I did not receive the inotify error before. My production repository is much larger than the simple tests I ran. 

Just for statistics:
* average internet speed:150 Mbps
* VM: digital ocean, ubuntu, 1cpu, 2GB mem, 50GB disk
